addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -569,9 +572,16 @@ public synchronized void playURI(String uri) {
     }
 
     private synchronized JsonArray getChannelGroups(final String channelType) {
-        JsonObject params = new JsonObject();
-        params.addProperty("channeltype", channelType);
-        JsonElement response = socket.callMethod("PVR.GetChannelGroups", params);
+        String method = "PVR.GetChannelGroups";
+        String hash = method + "#channeltype=" + channelType;
+        if (!REQUEST_CACHE.containsKey(hash)) {
```

Comment:
Could you use `REQUEST_CACHE.putIfAbsentAndGet` here?
https://github.com/eclipse/smarthome/blob/4204ce06bb28c28e5f711e720f87ef83beff2e27/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/cache/ExpiringCacheMap.java#L116